dashboard/test/integration/scripts_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class ScriptsTest < ActionDispatch::IntegrationTest
+  include LevelsHelper # for build_script_level_path
+
+  setup do
+    Rails.application.config.stubs(:levelbuilder_mode).returns true
+
+    # stub writes so that we dont actually make updates to filesystem
+    File.stubs(:write)
+
+    @teacher = create :authorized_teacher
+  end
+
+  test 'authorized teacher viewing csp1-2020' do
+    sign_in @teacher
+
+    @unit = create :script, name: 'csp1-2020'
+    @lesson_group = create :lesson_group, script: @unit
+    @lockable_lesson = create(:lesson, script: @unit, name: 'Assessment Day', lockable: true, lesson_group: @lesson_group, has_lesson_plan: false, absolute_position: 15, relative_position: 2)
+    @level_group = create(:level_group, :with_sublevels, name: 'assessment 1')
+    @lockable_level_group_sl = create(:script_level, script: @unit, lesson: @lockable_lesson, levels: [@level_group], assessment: true)
+
+    get build_script_level_path(@lockable_level_group_sl)
```

Comment:
after offline conversation... if these tests are just serving as backup for thorough manual testing, then it seems ok to test this, and this will likely have less churn than using an explicit url. OTOH if you are trying to get full test coverage via integration tests, then I would test each literal url explicitly.